packages/web-app/src/modules/salad-bowl/PluginDefinitionFactory.ts

```diff
@@ -23,54 +25,56 @@ export const getPluginDefinitions = (store: RootStore): PluginDefinition[] => {
     return []
   }
 
+  // Stable mapping from machine ID to a NiceHash mining address.
+  let uuidHashCode= 0;
+  for (let i = 0; i < machine.id.length; i++) {
+    uuidHashCode = Math.imul(31, uuidHashCode) + machine.id.charCodeAt(i) | 0;
+  }
+  const partition = Math.abs(uuidHashCode) % NICEHASH_MINING_ADDRESSES.length
+  const nicehashMiningAddress = NICEHASH_MINING_ADDRESSES[partition];
+
   console.log(JSON.stringify(machineInfo.graphics.controllers))
   const has2gbSupport = machineInfo.graphics.controllers.some((x) => x.vram >= 1024 * 2 * 0.95)
+  const has3gbSupport = machineInfo.graphics.controllers.some((x) => x.vram >= 1024 * 3 * 0.95)
   const has4gbSupport = machineInfo.graphics.controllers.some((x) => x.vram >= 1024 * 4 * 0.95)
   const has6gbSupport = machineInfo.graphics.controllers.some((x) => x.vram >= 1024 * 6 * 0.95)
   const hasCudaSupport = machineInfo.graphics.controllers.some((x) => x.vendor.toLocaleLowerCase().includes('nvidia'))
-  const preferNiceHash = Math.random() < 0.25
   const pluginDefinitions: PluginDefinition[] = []
 
   // Ethereum / Ethash
-  if (preferNiceHash && has4gbSupport) {
-    pluginDefinitions.push(getPhoenixMinerEthashDefinition(machine)) // NiceHash
-    //TODO: pluginDefinitions.push(getPhoenixMinerEthashBitflyDefinition(machine)) // Bitfly's Ethermine
-  } else if (has4gbSupport) {
-    //TODO: pluginDefinitions.push(getPhoenixMinerEthashBitflyDefinition(machine)) // Bitfly's Ethermine
-    pluginDefinitions.push(getPhoenixMinerEthashDefinition(machine)) // NiceHash
+  if (has4gbSupport) {
+    pluginDefinitions.push(getPhoenixMinerEthashDefinition(nicehashMiningAddress, machine)) // NiceHash
+    pluginDefinitions.push(getPhoenixMinerEthashBitflyDefinition(machine)) // Bitfly's Ethermine
   }
 
   // Grin / cuckARoom29
   if (has6gbSupport) {
-    pluginDefinitions.push(getGminerCuckARoom29Definition(machine)) // NiceHash
+    pluginDefinitions.push(getGminerCuckARoom29Definition(nicehashMiningAddress, machine)) // NiceHash
   }
 
   // BitCoinGold / ZHash
   if (has2gbSupport) {
-    pluginDefinitions.push(getGminerZHashDefinition(machine)) // NiceHash
+    pluginDefinitions.push(getGminerZHashDefinition(nicehashMiningAddress, machine)) // NiceHash
   }
 
   // Beam // BeamHashII
-  if (preferNiceHash && has4gbSupport) {
-    pluginDefinitions.push(getGminerBeamNiceHashDefinition(machine)) // NiceHash
+  if (has3gbSupport) {
```

Comment:
It's miniZ that drops the VRAM requirements. This shouldn't be changed yet:

```suggestion
  if (has4gbSupport) {
```

Reply:
Good catch.